### ISSUE TITLE
Trying to use the OAuthTokenProvider.exe to get the refresh token

### ISSUE THREAD

**Issue by jcus0006** [googleads/googleads-dotnet-lib#164]
At the last bit it redirects to a link that starts like this: http://localhost:8080/?code=4/AADn-oYSjybfr.....

I did not configure any endpoint to redirect to, as there was nowhere specified within the instructions: 

https://github.com/googleads/googleads-dotnet-lib/wiki/Using-OAuthTokenGenerator

Can someone kindly clarify if I missed some step?

**Comment by jcus0006** [comment]
I click OK, it redirects to the browser and the 8080 port is not found so I
get the error in the browser (as explained in the question), the app seems
to close on its own (it seems to disappear from the taskbar), so obviously
cannot revert to it in time to see any details.

**Comment by jcus0006** [comment]
I checked again, these are the steps:

1) In the Adwords API / Lib folder I open the OAuthTokenGenerator.exe file
2) I insert the client ID and secret and choose the Adwords API option
3) I click OK. The browser opens up with the account options. I choose the option I want.
4) The .exe window closes, and the browser tab redirects to this error: 

This site can’t be reached
localhost refused to connect.
Search Google for localhost 8080
ERR_CONNECTION_REFUSED

Any help would be appreciated as I am stuck.

**Comment by AnashOommen** [comment]
Is there another application running on port 8080? That's the most likely suspect, so OAuthTokenGenerator crashes when attempting to open port for local webserver.

I've pushed a newer version of the library, but that converts this application to a console application. The instructions are similar. Could you try with that version to see if the issue goes away?

**Comment by wmgdev** [comment]
I've just downloaded the new binary distribution v23.4.0

When I try run the new (dot net core) OAuthTokenGenerator console app, I get this.


```
C:\dev\Tutorials\Adwords\adwords_dotnet_lib_v23.4.0\lib>dotnet OAuthTokenGenerator.dll

Error:
  An assembly specified in the application dependencies manifest (OAuthTokenGenerator.deps.json) was not found:
    package: 'Google.Apis.Auth', version: '1.33.1'
    path: 'lib/netstandard1.3/Google.Apis.Auth.PlatformServices.dll'

```

**Comment by wmgdev** [comment]
In addition, if I clone the repository and build the project OAuthTokenGenerator from src.
The console app builds and runs ok.

I enter my client ID:
I enter my client secret:
I select yes for Adwords API
I select no for DFP API
I do not enter any additional OAuth2 scopes (I just press return)

A browser then opens to the following url:
[https://accounts.google.com/o/oauth2/v2/auth?access_type=offline&response_type=code&client_id=737887157553-kqkigk0gh9j711v4htoq5dpl250n4mo6.apps.googleusercontent.com&redirect_uri=http%3A%2F%2F127.0.0.1%3A60144%2Fauthorize%2F&scope](https://accounts.google.com/o/oauth2/v2/auth?access_type=offline&response_type=code&client_id=737887157553-kqkigk0gh9j711v4htoq5dpl250n4mo6.apps.googleusercontent.com&redirect_uri=http%3A%2F%2F127.0.0.1%3A60144%2Fauthorize%2F&scope)

My browser then get 'Error 400'
Error: invalid_request
Missing required parameter: scope

Request Details:
access_type=offline
response_type=code
client_id=737887157553-kqkigk0gh9j711v4htoq5dpl250n4mo6.apps.googleusercontent.com
redirect_uri=http://127.0.0.1:60144/authorize/
scope=

That's all I know ;-)

**Comment by wmgdev** [comment]
If I clone repo and build OAuthTokenGenerator from src it runs ok.

One small thing, when you have to enter yes/no, I initially just entered 'y' and 'n', force of habit lol
This meant my scope param was null and I was redirected to a 400 error page.
It might be nice to check for just 'y' and 'n' as well as ''yes' and 'no'

**Comment by AnashOommen** [comment]
Yeah, I was about to suggest to enter "yes" instead of "y". I'll enhance that part, and also publish the OAuthTokenGenerator as an exe instead of the DLL.

**Comment by AnashOommen** [comment]
Fixed in 23.5.0
